src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
+
+	// The product of the basket
+	static HashMap<String, Integer> map = new HashMap<String, Integer>();
+
+	static String[] localProducts = new String[]{"123", "222", "44", "657"};
+
+	// The fact that the basket has promo code
+	private static boolean codeDePromotion = false;
+
+	public void addProductToBasket(String product, Integer price, boolean isPromoCode) {
+		if (isPromoCode) {
+			codeDePromotion = true;
+		} else {
+			map.put(product, price);
+		}
+	}
+
+	public Long getBasketPrice(boolean inCents) {
+		Integer v = 0;
+		for (String s : map.keySet()) {
+			v += map.get(s);
+		}
+		if (codeDePromotion) {
+			v -= 100;
+		}
+		return inCents ? v * 100 : Long.valueOf(v);
+	}
```

Comment:
1. The prices are stored like ints or Longs, this is not optimal way to store money given the decimal precision it needs to be accurate with the amounts. Depending on the exactitude needed a double, BigDecimal and the Money API Java provides would be the way to go.

2. Variable v gives no information about what this variable is, price would be an improvement over it.

3. The return statement as a ternary reduces readability. It should be done in separate lines/statements and return the result.

4. This method return the price total of the basket and has a boolean to return in cents. This should be splitted in two methods getPrice and getPriceInCents.